clusterloader2/pkg/measurement/common/network/performance-measurement-helper.go

```diff
@@ -0,0 +1,137 @@
+		return err
+	}
+	if npm.protocol != protocolTCP && npm.protocol != protocolUDP && npm.protocol != protocolHTTP {
+		return fmt.Errorf("invalid protocol , supported ones are TCP,UDP,HTTP")
+	}
+	if err = npm.validatePodConfiguration(config); err != nil {
+		return err
+	}
+	npm.podRatioType = npm.getPodRatioType()
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) validatePodConfiguration(config *measurement.Config) error {
+	var err error
+	if npm.numberOfClients, err = util.GetInt(config.Params, "numberOfClients"); err != nil {
+		return err
+	}
+	if npm.numberOfServers, err = util.GetInt(config.Params, "numberOfServers"); err != nil {
+		return err
+	}
+	if npm.numberOfClients <= 0 || npm.numberOfServers <= 0 {
+		return fmt.Errorf("invalid number of client or server pods")
+	}
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) getPodRatioType() string {
+	if npm.numberOfClients == npm.numberOfServers && npm.numberOfClients == 1 {
+		return oneToOne
+	}
+	if (npm.numberOfClients > npm.numberOfServers) && npm.numberOfServers == 1 {
+		return manyToOne
+	}
+	if npm.numberOfClients == npm.numberOfServers {
+		return manyToMany
+	}
+	return invalid
+}
+
+func (npm *networkPerformanceMeasurement) populateTemplate(podPair uniquePodPair, index int) map[string]interface{} {
+	return map[string]interface{}{
+		"index":                index,
+		"clientPodName":        podPair.sourcePodName,
+		"serverPodName":        podPair.destinationPodName,
+		"clientPodIP":          podPair.sourcePodIP,
+		"serverPodIP":          podPair.destinationPodIP,
+		"protocol":             npm.protocol,
+		"duration":             npm.testDuration,
+		"clientStartTimestamp": npm.startTimeStampForTestExecution,
+		"numberOfClients":      1}
+}
+
+func getInformer(namespace string, k8sClient dynamic.Interface, gvr schema.GroupVersionResource) (cache.SharedInformer, error) {
+	optionsModifier := func(options *metav1.ListOptions) {}
+	tweakListOptions := dynamicinformer.TweakListOptionsFunc(optionsModifier)
+	dynamicInformerFactory := dynamicinformer.NewFilteredDynamicSharedInformerFactory(k8sClient, 0, namespace, tweakListOptions)
+	informer := dynamicInformerFactory.ForResource(gvr).Informer()
+	return informer, nil
+}
+
+type float64Slice []float64
+
+func (p float64Slice) Len() int           { return len(p) }
+func (p float64Slice) Less(i, j int) bool { return p[i] < p[j] }
+func (p float64Slice) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
+
+func getPercentile(values float64Slice, percentile float64) float64 {
+	var score float64 = 0
+	size := len(values)
+	if size <= 0 {
+		return score
+	}
+
+	sort.Sort(values)
+	pos := percentile * float64(size+1)
+	if pos < 1.0 {
+		score = values[0]
+	} else if pos >= float64(size) {
+		score = values[size-1]
+	} else {
+		lower := values[int(pos)-1]
+		upper := values[int(pos)]
+		score = lower + (pos-math.Floor(pos))*(upper-lower)
```

Comment:
Is intrapolation really what we want here? Don't we want to simply return lower/upper bound of it?


FTR: https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/pkg/measurement/util/latency_metric.go#L94

Reply:
I think interpolation gives more realistic data when we go with larger sets. So i think , we should retain this.

Reply:
With larger sets, it actually doesn't really matter. You're introducing complexity for unclear gain.

Reply:
ok modified